src/grouping-and-mapping/clients/PropertiesClient.ts

```diff
@@ -0,0 +1,125 @@
+import { IPropertiesClient } from "../interfaces/IPropertiesClient";
+import { ECPropertyReference, Property, PropertyContainer, PropertyList, PropertyModify } from "../interfaces/Properties";
+import { EntityListIteratorImpl } from "../../common/iterators/EntityListIteratorImpl";
+import { Collection, getEntityCollectionPage } from "../../common/iterators/IteratorUtil";
+
+export class PropertiesClient extends OperationsBase implements IPropertiesClient {
+  private _propertiesUrl = `${this.groupingAndMappingBasePath}/datasources/imodel-mappings`;
+
+  public async createProperty(accessToken: AccessToken, mappingId: string, groupId: string, property: PropertyModify): Promise<Property> {
+    if (!this.isSimpleIdentifier(property.propertyName)) {
+      throw new RequiredError(
+        "propertyName",
+        "Field propertyName was invalid.",
+      );
+    }
+
+    if(property.ecProperties)
+      for(const ecProperty of property.ecProperties) {
+        if (!this.isValidECProperty(ecProperty)) {
+          throw new RequiredError(
+            "ecProperties",
+            "Field ecProperties was invalid.",
+          );
+        }
+      }
+
+    const url = `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties`;
+    const requestOptions: RequestInit = this.createRequest("POST", accessToken, JSON.stringify(property));
+    return (await this.fetchJSON<PropertyContainer>(url, requestOptions)).property;
+  }
+
+  public async deleteProperty(accessToken: AccessToken, mappingId: string, groupId: string, propertyId: string): Promise<Response> {
+    const url = `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties/${encodeURIComponent(propertyId)}`;
+    const requestOptions: RequestInit = this.createRequest("DELETE", accessToken);
+    return this.fetchJSON<Response>(url, requestOptions);
+  }
+
+  public async getProperty(accessToken: string, mappingId: string, groupId: string, propertyId: string): Promise<Property> {
+    const url = `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties/${encodeURIComponent(propertyId)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return (await this.fetchJSON<PropertyContainer>(url, requestOptions)).property;
+  }
+
+  public async getProperties(accessToken: string, mappingId: string, groupId: string, top?: number | undefined): Promise<PropertyList> {
+    if(!this.topIsValid(top)) {
+      throw new RequiredError(
+        "top",
+        "Parameter top was outside of the valid range [1-1000]."
+      );
+    }
+
+    const url = top ? `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties?$top=${top}`
+      : `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties`;
```

Comment:
The construction of this url looks the same as in `getProperty`. Try making a helper function with top being `optional`.

Reply:
Helper function created, which makes me think that groups client could also use one. And to take it further, maybe a general endpoint url constructor would be useful since `most` grouping-and-mapping clients follow the same pattern.